### ISSUE TITLE
Extract text content from PDFs

### ISSUE THREAD

**Issue by Treora** [WebMemex/webmemex-extension#27]
Mainly just making a cleanup of Oliver's code: https://github.com/oliversauter/pdf2text4extensions

**Comment by bohrium272** [comment]
I'd like to take this up. Please describe further.

**Comment by Treora** [comment]
Great. I imagine this to be a separable module, though possibly developed within this repo for now, e.g. in `/src/pdf-analysis`.

As we quickly discussed, there may be two situations to consider: one is where the user is visiting the document, one is where we just have the url and should download it first. I am not sure about the possibilities in the first case, we might be required to download it again if we cannot access it otherwise. Anyway, we would probably end up with a [Blob](https://developer.mozilla.org/en-US/docs/Web/API/Blob) containing the document.

So let's start with making a single function that, given this pdf blob, returns an object with its plain text and metadata, for example like so:
```
>>> analysePdf(myPdfBlob)
{
  plainText: "bla bla blabla...",
  title: "Bla.",
  numPages: 3,
  ...
}
```

We'll probably use the core of [PDF.js](https://mozilla.github.io/pdf.js/) in the same way [Oliver's code](https://github.com/oliversauter/pdf2text4extensions) does, though I am not sure if we can import just the core or if it only comes along with the rendering part. Perhaps have a look at its docs and examples.

**Comment by oliversauter** [comment]
I talked to @vaibhav, he wanted to take this task on. Is that still valid Vaibhav?

@Treora I suggest adding him as a contributor, so we can assign the task.

**Comment by blackforestboi** [comment]
I talked to @vaibsharma, he wanted to take this task on. Is that still valid Vaibhav?

@Treora I suggest adding him as a contributor, so we can assign the task.

**Comment by vaibsharma** [comment]
Yes I want to take this task .

**Comment by blackforestboi** [comment]
ok great. Happy Hacking!

**Comment by Treora** [comment]
@vaibsharma / @oliversauter: do I understand plans changed again? For me this feature does not have a high priority and nothing is blocked by it, so no problem if it moves to the waiting list again, but if somebody wants to implement it I am glad to assist.

**Comment by blackforestboi** [comment]
@vaibsharma is having his midterm exams until 11 March, so wont be able to take it on anytime soon and there are other people who want to take it on...soon coming :)

**Comment by vaibsharma** [comment]
I pushed whatever I was able to commit . Please have a look at #43 . Rest I will do after midsems :) . Thanks

**Comment by blackforestboi** [comment]
I am sorry I might have mixed up some things...so many new people in the last days. :/

It appears that @vaibsharma was working on this!

**Comment by RajPratim21** [comment]
Hi, I am taking a try on it.

**Comment by vaibsharma** [comment]
Sorry @Treora  @oliversauter  for the delay . I was busy with my university exams .
Is this issue is not mandatory for now or someone is working on it ?
If not .. then can I again start with it ?  
Edit : I didn't see your comment in the PR page . Okay then I try to take on the another issue .
Thanks

**Comment by Treora** [comment]
Another overripe issue. Was fixed in #88 (merged in d59ce46).
